payment/payment.go

```diff
@@ -40,7 +40,8 @@ func FromPublicKey(pubkey *btcec.PublicKey, network *network.Network) *Payment {
 func FromPayment(payment *Payment) *Payment {
 	buf := payment.Script
 	hash := hash160(buf)
-	return &Payment{payment.Network, payment.PublicKey, hash, nil, nil, nil}
+	redeem := &Payment{payment.Network, payment.PublicKey, hash, nil, nil, nil}
+	return &Payment{nil, nil, nil, nil, redeem, nil}
```

Comment:
I rather return something like:
```
return &Payment{payment.Network, nil, hash, nil, payment, nil}
```

Reply:
Also this way you are rewriting the Hash that it's already present in given payment in the redeem field. 

Eg. In case of `payment` having already a `Hash` beacuse it's a `pay to witness pubkey hash` this way you are reqriting it with an extra hash160 that's only neede by ScriptHash construction.

I'm right @altafan it's better preserve the original given payment in the `Redeem` field?

Reply:
Yes, the incoming `payment` doesn't and shouldn't be modified but just used to produce the hash of the `*Payment` to return, as it's already done, and also included in it in the `Redeem` field.

PS: I think my example is not the best option because by using the same `payment` pointer  for the return value of the `FromPayment` method, this could lead to unexpected behaviors if for any reason the `payment` pointer is deleted. I think it would be better to make a copy and use a pointer to a different memory allocation.